### PR TITLE
Sleep Timer: Add the option to select the number of chapters

### PR DESCRIPTION
**Pull request by mebarbosa**
## Description
- This adds the ability to select multiple chapters for sleep time
- Important: Since we have episodes with chapters and episodes without chapters, we are going to treat an episode without chapters as a single chapter.
- In this version, every time a user switches to another chapter or episode, we will decrease the number of chapters left before sleep.

### End of Chapters
1. Run the app
2. Add 5/6 episodes to Up next. Try to add episodes that contain chapters and others that don't
3. Open the full player, tap the sleep timer icon (zZz)
4. Select X chapters for sleep timer
5. ✅ Ensure 🔵 Tracked: player_sleep_timer_enabled, Properties: {"time":"end_of_chapter","number_of_chapters": X
6. Tap the sleep timer icon (zZz) again
7.  ✅  You should see `Sleeping in X chapters` message and the cancel button
8. If your episodes has chapter, try to move to the next chapter
9. If you check the _sleep in_ message, it should be decreased by 1 since you changed the chapter
10. If your episodes does not have chapter, try to move to the next episode
11. If you check the _sleep in_ message, it should be decreased by 1 since you changed the episode
12. Once your sleep timer is triggered and you still have episodes on Up next you can tap on `Play` button again
13. ✅ Automatic sleep time should be triggered with X chapters set
14. ✅  Ensure  🔵 Tracked: player_sleep_timer_restarted, Properties: {"time":"end_of_chapter","number_of_chapters":X

### End of Chapters + Up Next
1. Run the app
2. Go Settings -> General -> Have `Play up next episode on tap` enabled
3. Add 5/6 episodes to Up next. Try to add episodes that contain chapters and others that don't
4. Open the full player, tap the sleep timer icon (zZz)
5. Select 5 chapters for sleep timer
6. Tap the sleep timer icon (zZz) again
7. ✅  You should see `Sleeping in 5 chapters` message and the cancel button
8. Tap to another episode from Up Next
9. Tap the sleep timer icon (zZz) again
10. ✅  You should see `Sleeping in 4 chapters` message and the cancel button

### Shake Device
1. Have sleep time set for chapters
2. Shake your device
3. ✅ Ensure the sleep timer was not restarted

### Auto Play
1. Add some episodes to Up next
2. Do not set sleep timer
3. Play an episode
4. Scroll to almost the end of the episode
5. Wait for episode to finish
6. ✅ Ensure the next episode from Up Next starts playing

## Screenshots or Screencast 
[Screen_recording_20240426_140654.webm](https://github.com/Automattic/pocket-casts-android/assets/42220351/2968d9e8-570e-4e95-9bc7-cf144dc71ea1)

## Checklist
- [x] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

File: modules/services/localization/src/main/res/values/strings.xml
Comment (mebarbosa) [inline modules/services/localization/src/main/res/values/strings.xml:356]: Hey @david-gonzalez-a8c I am displaying this `Sleep timer stopped at the end of your chapter` message in the toast when the sleep timer has stopped for end of chapters. What do you think? Do you have another suggestion?

Comment (mebarbosa) [reply modules/services/localization/src/main/res/values/strings.xml:356]: cc: @leandroalonso

**Comment by CookieyedCodes** [comment]
Just my two cents, When I was doing my UX for chapters Initially I had it so that the end of episode and chapters would be interchangeable text wise depending on weather the podcast had chapters or not, this was eventually replaced with a grayed out "their are no chapters in this podcast" as I had worked on the UI enough that end of episode was allways included in the list, tbh I prefer the former if it's possible I love the end of episode timer tho it might get tangled with the repeater if it's being shared with the chapter timer🤔 

I also had it so that the user couldn't go beyond the maximum number of chapters so the number was kind of dynamic, I also had it that the chapter timer would reset to 1 every episode because while that might be annoying it would help with NaN, this is also why I had a +1 & -1 on the timer screen so the user would be able to control if they wanted to change their mind with chapters quickly, happy this is being added tho :)

File: modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackManager.kt
Comment (ashiagr) [inline modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackManager.kt:2556]: Looks like typo: 

Can we rename the following keys wherever they are used?
`episodeUui` -> `episodeUuid`
`chapterUui` -> `chapterUuid`

**Review by ashiagr — APPROVED** [review]
LGTM based on the current design.
Any design updates can be addressed during beta.

File: modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackManager.kt
Comment (mebarbosa) [reply modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackManager.kt]: Nice catch! Fixed here: [6006443](https://github.com/Automattic/pocket-casts-android/pull/2115/commits/600644324df63f505b8451df51d3e7fd7d60875b)

**Comment by mebarbosa** [comment]
> Just my two cents, When I was doing my UX for chapters Initially I had it so that the end of episode and chapters would be interchangeable text wise depending on weather the podcast had chapters or not, this was eventually replaced with a grayed out "their are no chapters in this podcast" as I had worked on the UI enough that end of episode was allways included in the list, tbh I prefer the former if it's possible I love the end of episode timer tho it might get tangled with the repeater if it's being shared with the chapter timer🤔
> 
> I also had it so that the user couldn't go beyond the maximum number of chapters so the number was kind of dynamic, I also had it that the chapter timer would reset to 1 every episode because while that might be annoying it would help with NaN, this is also why I had a +1 & -1 on the timer screen so the user would be able to control if they wanted to change their mind with chapters quickly, happy this is being added tho :)



> Just my two cents, When I was doing my UX for chapters Initially I had it so that the end of episode and chapters would be interchangeable text wise depending on weather the podcast had chapters or not, this was eventually replaced with a grayed out "their are no chapters in this podcast" as I had worked on the UI enough that end of episode was allways included in the list, tbh I prefer the former if it's possible I love the end of episode timer tho it might get tangled with the repeater if it's being shared with the chapter timer🤔
> 
> I also had it so that the user couldn't go beyond the maximum number of chapters so the number was kind of dynamic, I also had it that the chapter timer would reset to 1 every episode because while that might be annoying it would help with NaN, this is also why I had a +1 & -1 on the timer screen so the user would be able to control if they wanted to change their mind with chapters quickly, happy this is being added tho :)

@CookieyedCodes I liked your `their are no chapters in this podcast` suggestion. Right now we are trying to release this feature as simple as possible so that's why we are not checking some others cases. I will share this with @david-gonzalez-a8c to discuss before merge this PR. I also updated to `7.64` so we have time to work on it.

Just to let you know that we are going to do some redesigns around this feature so we should see some UI improvements on this area too.

Thank you again for all your feedbacks!

**Comment by CookieyedCodes** [comment]
No bothers you can tell I've been thinking about it for a long time :) 😂, tbh I kind of prefer the latter just because I prefer to have more somewhat permanent end of episode button, David has my UX suggestions but their are my exploration under my niggles thread that links out to a conversation I had with Phil under discussions 😉

Looking forward to actual implementation tho:)

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">This PR is tagged with <code>do not merge</code> label(s).</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: modules/services/localization/src/main/res/values/strings.xml
Comment (mebarbosa) [reply modules/services/localization/src/main/res/values/strings.xml:357]: We are going to keep this message. I confirme with David on slack

**Pull request merged**
